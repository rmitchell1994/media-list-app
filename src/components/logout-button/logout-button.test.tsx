import React from "react";
import { useMutation } from "react-query";
import { act, fireEvent, render, screen } from "../../../test/test-utils";
import LogoutButton from "./logout-button";

jest.mock("react-query", () => ({
  ...jest.requireActual<{ module: "react-query" }>("react-query"),
  useMutation: jest.fn().mockReturnValue({
    mutate: jest.fn(),
  }),
}));

describe("Logout Button", () => {
  it("shoud render logout button", () => {
    const { asFragment } = render(<LogoutButton />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should call mutate on click", () => {
    render(<LogoutButton />);

    act(() => {
      fireEvent.click(screen.getByText("Log Out"));
    });

    expect(useMutation).toHaveBeenCalled();
  });
});
