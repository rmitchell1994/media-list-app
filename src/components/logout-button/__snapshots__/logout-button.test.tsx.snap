// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Button shoud render logout button 1`] = `
<DocumentFragment>
  <a
    class="sc-bdfBwQ jjEszW"
    data-testid="logout-button"
  >
    Log Out
  </a>
</DocumentFragment>
`;
